addons/mail/static/src/discuss/core/common/avatar_stack.js

```diff
@@ -0,0 +1,77 @@
+import { Component, useEffect, useState } from "@odoo/owl";
+import { _t } from "@web/core/l10n/translation";
+
+/**
+ * @typedef {Object} Props
+ * @prop {(expanded: boolean) => void} [onToggle] Callback to be invoked when
+ * the avatar stack expends or collapses.
+ * @prop {(persona: import("models").Persona) => string} [getAvatarClass]
+ * Function used to determine extra classes for the avatars.
+ * @prop {Array} personas List of personas to display in the stack.
+ * @prop {"horizontal"|"vertical"} [direction] Determine the direction of the
+ * stack.
+ * @prop {boolean} [expandable] Whether the stack can expend to show every
+ * persona avatar and name in column.
+ * @prop {number} [forceExpand] Number to increment to force the stack to
+ * expend.
+ * @prop {number} [max] Maximum number of personas to display in the stack. An
+ * hidden count will be displayed if there are more personas than max.
+ * @prop {number} [size] Size of the avatars, in pixel.
+ * @extends {Component<Props, Env>}
+ */
+export class AvatarStack extends Component {
+    static template = "mail.AvatarStack";
+    static props = {
+        direction: { type: String, optional: true },
+        expandable: { type: Boolean, optional: true },
+        forceExpand: { type: Number, optional: true },
+        getAvatarClass: { type: Function, optional: true },
+        max: { type: Number, optional: true },
+        onToggle: { type: Function, optional: true },
+        personas: Array,
+        size: { type: Number, optional: true },
+        slots: Object,
+    };
+    static defaultProps = {
+        getAvatarClass: () => "",
+        max: 4,
+        onToggle: () => {},
+        size: 24,
+        direction: "horizontal",
+    };
+
+    setup() {
+        super.setup(...arguments);
+        this.state = useState({ expanded: false });
+        useEffect(
+            (expanded) => void this.props.onToggle(expanded),
+            () => [this.state.expanded]
+        );
+        useEffect(
+            (forceExpand, expandable) => {
```

Comment:
TBH this is questionable. The component displays personas, it could very much have a compact and expanded form. This is not a big deal for now though as we only use it in once place but I think it makes sense; but not a big deal, so for now I'll just move it to the parent component and we'll see whether the need arises or not in the future.

As for the "compact"/"expanded" wording. I don't think that "expanded is more intuitive". What is more intuitive is how the component behave (it's a stack, it's compact) and how we can customize it, here "expand" it. But once again not a big deal so I can adapt it :)

Reply:
The "expand" wording comes mostly from `AvatarStack` managing both states, which feels off. The listing of call participants should not in the concern of avatar stack but instead a concern of call participants in sidebar.

So avatar stack will lack the mode, instead this moved to discuss sidebar call participants. Then this is more subjective but it's usually more natural in "expanded" by default and then have conditional on "compact". This is because expanded gives a better macro view of all features, and "compact" usually comes as a custom look to reduce immediately availability of information for less distraction, thus this feels like a customisation on top of plain listing of participants. Technical & function complexity of each mode also contributes to decision, plain listing of participants (non-compact) feels less complex.